src/index.ts

```diff
@@ -1,15 +1,42 @@
 import path from 'path'
 import fs from 'fs-extra'
 import globby from 'globby'
+// @ts-ignore
 import ware from 'ware'
 
+type Middleware = (ctx: Majo) => any
+type Glob = string | string[]
+type TransformFn = (contents: string) => Promise<string> | string
+
+interface Files {
+  [relativePath: string]: File
+}
+
+interface File {
+  path?: string,
+  contents: Buffer,
+  stats: fs.Stats,
+  [key: string]: any
+}
+
+interface StatCache {
+  [path: string]: fs.Stats
+}
+
 class Majo {
+  middlewares: Middleware[]
+  meta: any
+  baseDir: string
+  sourcePatterns: Glob
```

Comment:
It's even better to add `private/public/protected` declaration at each field. I guess the `sourcePatterns` and `dotFiles` are private fields 😅